bottlerocket/samples/RUNBOOK.md

```diff
@@ -1,438 +1,57 @@
-  --name "/aws/service/bottlerocket/${VARIANT}/${ARCHITECTURE}/latest/image_id" \
-  --query Parameter.Value --output text)
+## Running a test file
 
-cli add-secret map  \
- --name "aws-creds" \
- "ACCESS_KEY_ID=${ACCESS_KEY_ID}" \
- "SECRET_ACCESS_KEY=${SECRET_ACCESS_KEY}"
+A test file can be run by executing a `cargo make run <test-name>` command. The following example will run an ECS test.
 
-eval "cat > ${OUTPUT_FILE} << EOF
-$(< kind/sonobuoy-test.yaml)
-EOF
-" 2> /dev/null
+```bash
+cargo make run ecs-test
 ```
 
-### Workload Testing on `aws-k8s` Variants
+This command will generate the test file first, then run it on a Testsys cluster. The status can be checked using `cli status`.
 
-_Note_: An example of a workload test that can be used with the `k8s-workload-agent` is the `nvidia-smoke` test. See the `nvidia-smoke` [README.md](../tests/workload/nvidia-smoke/README.md) for instructions on how to build and use its image.
+_Note_: `cargo make run-samples` can be used to run all samples in the directory specified by `CLUSTER_TYPE`. This command cannot accept user-defined values for `CLUSTER_NAME`, `GPU`, `METADATA_URL`, `OVA_NAME`, `BOTTLEROCKET_AMI_ID`, `INSTANCE_TYPES`, or `VARIANT`.
 
-```bash
-CLUSTER_NAME="x86-64-aws-k8s-124-nvidia"
-OUTPUT_FILE="${CLUSTER_NAME}.yaml"
-VARIANT="aws-k8s-1.24-nvidia"
-ARCHITECTURE="x86_64"
-AGENT_IMAGE_VERSION=$(cli --version | sed -e "s/^.* //g")
-K8S_WORKLOAD_AGENT_IMAGE_URI="public.ecr.aws/bottlerocket-test-system/k8s-workload-agent:v${AGENT_IMAGE_VERSION}"
-EKS_RESOURCE_AGENT_IMAGE_URI="public.ecr.aws/bottlerocket-test-system/eks-resource-agent:v${AGENT_IMAGE_VERSION}"
-EC2_RESOURCE_AGENT_IMAGE_URI="public.ecr.aws/bottlerocket-test-system/ec2-resource-agent:v${AGENT_IMAGE_VERSION}"
-ASSUME_ROLE="~"
-AWS_REGION="us-west-2"
-WORKLOAD_TEST_NAME=
-WORKLOAD_TEST_IMAGE_URI=
-GPU="true"
-INSTANCE_TYPES=$(if [ $GPU = "true" ]; then echo "[\"g4dn.xlarge\"]"; else echo "[\"m5.large\"]"; fi)
+## Cleaning the output folder
 
-BOTTLEROCKET_AMI_ID=$(aws ssm get-parameter \
-  --region ${AWS_REGION} \
-  --name "/aws/service/bottlerocket/${VARIANT}/${ARCHITECTURE}/latest/image_id" \
-  --query Parameter.Value --output text)
+The output folder can be cleaned by executing `cargo make clean-output`.
 
-cli add-secret map  \
- --name "aws-creds" \
- "ACCESS_KEY_ID=${ACCESS_KEY_ID}" \
- "SECRET_ACCESS_KEY=${SECRET_ACCESS_KEY}"
+## Special considerations
 
-eval "cat > ${OUTPUT_FILE} << EOF
-$(< kind/k8s-workload-test.yaml)
-EOF
-" 2> /dev/null
-```
-
-### Conformance Testing on `vmware-k8s` Variants
+### Workload tests
 
-This codeblock assumes that your vSphere config file has been sourced. Specifically, the variables `GOVC_USERNAME`, `GOVC_PASSWORD`, `GOVC_DATACENTER`, `GOVC_DATASTORE`, `GOVC_URL`, `GOVC_NETWORK`, `GOVC_RESOURCE_POOL`, and `GOVC_FOLDER` need to be populated.
+When creating or running a workload test file, the variables `WORKLOAD_TEST_NAME` and `WORKLOAD_TEST_IMAGE_URI` need to be set by the user.
```

Comment:
Is there a check for this in the Makefile? Is there a way to check the sample for values that weren't replaced?

Reply:
A check to make sure that these variables have values can be added to the Makefile.